deepinv/datasets/urban100.py

```diff
@@ -0,0 +1,135 @@
+                   -- Urban100_SR.zip
+
+    For each image in the dataset, several variations of the same image exist.
+    For this class, we are dealing with only high resolution images in `image_SRF_4` folder.
+    For more information about the different versions, look at `readme.txt`.
+
+    >>> root = "/path/to/dataset/Urban100"
+    ... dataset = Urban100HR(root=root, download=True)  # will download dataset at root
+    ... dataset.check_dataset_exists()                  # check that raw data has been downloaded correctly
+
+    :param str root: Root directory of dataset. Directory path from where we load and save the dataset.
+    :param bool download: If True, downloads the dataset from the internet and puts it in root directory.
+        If dataset is already downloaded, it is not downloaded again. Default at False.
+    :param callable, optional transform: A function/transform that takes in a PIL image
+        and returns a transformed version. E.g, ``torchvision.transforms.RandomCrop``
+    """
+
+    zipfile_urls = {
+        "Urban100_SR.zip": "https://uofi.box.com/shared/static/65upg43jjd0a4cwsiqgl6o6ixube6klm.zip",
+    }
+
+    # for integrity of downloaded data
+    checksums = {
+        "image_SRF_2": "7a69080e004abff22afea2520f7d7e83",
+        "image_SRF_4": "7c4479537ef7bf42270cf663205a136b",
+    }
+
+    def __init__(
+        self,
+        root: str,
+        download: bool = False,
+        transform: Callable = None,
+    ) -> None:
+        self.root = root
+        self.transform = transform
+        self.img_dir = os.path.join(self.root, "image_SRF_4")
+
+        # download dataset, we check first that dataset isn't already downloaded
+        if not self.check_dataset_exists():
+            if download:
+                if not os.path.isdir(self.root):
+                    os.makedirs(self.root)
+                if os.path.exists(self.img_dir):
+                    raise ValueError(
+                        f"""
+                    The image folder already exists,
+                    thus the download is aborted.
+                    Please set `download=False`
+                    OR remove `{self.img_dir}`."""
+                    )
+
+                for filename, url in self.zipfile_urls.items():
+                    # download zipfile from the Internet and save it locally
+                    download_zipfile(
+                        url=url,
+                        save_path=os.path.join(self.root, filename),
+                    )
+                    # extract local zipfile
+                    extract_zipfile(os.path.join(self.root, filename), self.root)
+
+                    if self.check_dataset_exists():
+                        print("Dataset has been successfully downloaded.")
+            # stop the execution since the dataset is not available and we didn't download it
+            else:
+                raise RuntimeError(
+                    f"""
```

Comment:
Here be careful that this prints without new lines.

Reply:
it prints a new lines, i didn't found an elegant way to avoid the new line and having a clean str msg

Reply:
I think it is better to remove the new lines (eg torch's standard printing doesn't use newlines)

Reply:
ok